src/app/frontend/deploy/deploy_settings_controller.js

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {stateName as replicasetliststate} from 'replicasetlist/replicasetlist_state';
+
+/**
+ * Controller for the deploy using settings directive.
+ *
+ * @final
+ */
+export default class DeploySettingsController {
+  /**
+   * @param {!angular.Scope} $scope
+   * @param {!angular.$log} $log
+   * @param {!ui.router.$state} $state
+   * @param {!angular.$resource} $resource
+   * @ngInject
+   */
+  constructor($scope, $log, $state, $resource) {
```

Comment:
Quick comment: (almost) never use $scope, unless you're doing something that is application wide. All directives and views should have `bindToController=true` and `controllerAs:'ctrl'`.

Instead of communicating by events you can use plain old directive scope parameters. Communicating by events is hard to read and understand, so avoid it when you can and opt for explicit callbacks. E.g., make this directive look like this (deploy.html):
```
<deploy-from-settings deploy="ctrl.deployFn" />

Then the deploy_controller.js would have
/** @export {function():angular.$q.Promise} */
this.deployFn 

The deployFn would return a deploy promise. With the promise you can update this.isDeployInProgress_ easily. 
```

Does this make sense? What do you think?